pkg/cluster/pg.go

```diff
@@ -45,25 +46,42 @@ func (c *Cluster) databaseAccessDisabled() bool {
 	return !c.OpConfig.EnableDBAccess
 }
 
-func (c *Cluster) initDbConn() (err error) {
+func (c *Cluster) initDbConn() error {
 	c.setProcessName("initializing db connection")
-	if c.pgDb == nil {
-		conn, err := sql.Open("postgres", c.pgConnectionString())
-		if err != nil {
-			return err
-		}
-		c.logger.Debug("new database connection")
-		err = conn.Ping()
-		if err != nil {
+	if c.pgDb != nil {
+		return nil
+	}
+
+	conn, err := sql.Open("postgres", c.pgConnectionString())
+	if err != nil {
+		return err
+	}
+
+	c.logger.Debug("new database connection")
+	err = retryutil.Retry(0, constants.PostgresConnectRetryTimeout,
+		func() (bool, error) {
+			err := conn.Ping()
+			if err == nil {
+				return true, nil
+			}
+
 			if err2 := conn.Close(); err2 != nil {
 				c.logger.Errorf("error when closing PostgreSQL connection after another error: %v", err2)
+				return false, err2
 			}
-			return err
-		}
 
-		c.pgDb = conn
+			if _, ok := err.(*net.OpError); ok {
+				return false, nil
```

Comment:
Why would we hide this error?

Reply:
you mean we should log it?

Reply:
yes, logging it would help us in finding the root cause.